cmd/tink-worker/main.go

```diff
@@ -2,18 +2,56 @@ package main
 
 import (
 	"fmt"
+	"math/rand"
 	"os"
+	"time"
 
+	"github.com/tinkerbell/tink/cmd/tink-worker/client/tink"
 	"github.com/tinkerbell/tink/cmd/tink-worker/cmd"
+	"github.com/tinkerbell/tink/protos/workflow"
+	"google.golang.org/grpc"
 )
 
 func main() {
-	cmdlineFlags, err := cmd.CollectCmdlineFlags(os.Args[1:])
+	// parse and validate command-line flags and required env vars
+	flagEnvSettings, err := cmd.CollectFlagEnvSettings(os.Args[1:])
 	if err != nil {
 		fmt.Println("Error:", err)
 		os.Exit(1)
 	}
 
-	// Remove me: this is just here to prevent linter issues
-	fmt.Printf("Version flag is %v\n", cmdlineFlags.Version)
+	// establish connection to tink server
+	creds, err := tink.ObtainServerCreds(flagEnvSettings.TinkServerURL)
+	if err != nil {
+		fmt.Println("Error obtaining tink server TLS credentials:", err)
+		os.Exit(1)
+	}
+	// Here we retry failed connections to tink-server using a randomized interval between attempts.
+	// tink-worker is a daemon process, so we do not exit here.
+	var conn *grpc.ClientConn
+	for {
+		conn, err = tink.EstablishServerConnection(flagEnvSettings.TinkServerGRPCAuthority, creds)
+		if err == nil {
+			break
+		}
+
+		fmt.Println("Error establishing connection to tink-server:", err)
+		// sleep a randomized amount of time before reconnecting to avoid thundering herds
+		// TODO: we may want to make this configurable via a cmdline option?
+		rand.Seed(time.Now().UnixNano())
+		sleepSeconds := rand.Intn(120) // 2 minutes (120 seconds)
+		fmt.Printf("Sleeping %d seconds before attempting to re-connect...\n", sleepSeconds)
+		time.Sleep(time.Duration(sleepSeconds) * time.Second)
+	}
+
+	workflowClient := workflow.NewWorkflowServiceClient(conn)
+
+	// Remove me: this is just here during developent to prevent linter issues
```

Comment:
This needs a `TODO`, or needs to be removed.

Reply:
Agreed, fixed.